### PR TITLE
adc: add pinctrl support to adc sam0 driver

### PR DESCRIPTION
**Pull request by erikkallen**
Update the sam0 adc driver with pinctrl support

Signed-off-by: Erik Kallen <info@erikkallen.nl>

### CODE REVIEW

File: dts/bindings/adc/atmel,sam0-adc.yaml [hunk 1]
```diff
@@ -43,6 +43,18 @@ properties:
...
         - 14
+    pinctrl-0:
+      type: phandles
+      description: |
+        PIO pin configuration for adc pin and channel.  We expect that
+        the phandles will reference pinctrl nodes.  These nodes will
+        have a nodelabel that matches the Atmel SoC HAL defines and
+        be of the form p<port><pin><periph>_<chan>_<signal>.
+
+        For example the I2C0 on SAMD21 would be
```
Comment (IoannisPZP) [inline dts/bindings/adc/atmel,sam0-adc.yaml]: I2C0?

File: dts/bindings/adc/atmel,sam0-adc.yaml [hunk 2]
```diff
@@ -43,6 +43,18 @@ properties:
...
+           pinctrl-0 = <&pa9d_adc_chan19>;
```
Comment (IoannisPZP) [inline dts/bindings/adc/atmel,sam0-adc.yaml:55]: Is an "_" missing between chan and 19 or is the syntax stated in line 52 wrong?

**Review by IoannisPZP — CHANGES_REQUESTED** [review]
Looks good further

File: dts/bindings/adc/atmel,sam0-adc.yaml (hunk 2 above)
Comment (erikkallen) [reply dts/bindings/adc/atmel,sam0-adc.yaml:55]: Syntax is now corrected

File: dts/bindings/adc/atmel,sam0-adc.yaml (hunk 1 above)
Comment (erikkallen) [reply dts/bindings/adc/atmel,sam0-adc.yaml]: Copy paste error fixed

File: dts/bindings/adc/atmel,sam0-adc.yaml (hunk 2 above)
Comment (erikkallen) [reply dts/bindings/adc/atmel,sam0-adc.yaml:55]: It still is confusing but p a=port 9=pin d=periph inst=adc and chan=chan19

Comment (IoannisPZP) [reply dts/bindings/adc/atmel,sam0-adc.yaml:55]: would have made it `_chan<channel_number>` but well its fine since there is an example

**Pull request merged**